### ISSUE TITLE
No warning for unescaped newlines in $() calls

### ISSUE THREAD

**Issue by refola** [koalaman/shellcheck#390]
This code is invalid.

```sh
#!/bin/sh
f() { echo "$@"; }
x=$(f arg1
    arg2)
echo "$x"
```

It looks like it's meant to set `x` to the output of `f arg1 arg2`, but the newline between `arg1` and `arg2` causes a `command not found` error for `arg2`. Shellcheck doesn't catch this.

The fix is simple: Add a ` \` (space, backslash) to the end of `arg1` (generally, every line before the closing `)`), allowing the multi-line `$()` call to continue as intended.

**Comment by koalaman** [comment]
This would be useful in a number of contexts, but how do you tell them apart from legitimate cases like

    x=$(f "Dear $name"
        cat email_body
        f "Yours truly")

    echo "$x" | mail someone@example.com

**Comment by refola** [comment]
I honestly didn't know that the `$()` syntax handled multiple commands, even though the `command not found` error should have made it really obvious.

The fix then becomes requiring either a `\` or a `;` at the end of each line in a `$()` expression. But at that point you might as well do the same everywhere else, which would probably cause more annoyance than it would solve.

Bash already gives a usable error message when this happens, and "fixing" it in Shellcheck would require more annoyance than it's worth. Can you close this as "invalid"/"wontfix" or similar?
